src/Database/PostgreSQL/PQTypes/SQL/Builder.hs

```diff
@@ -239,15 +246,15 @@ data SqlUpdate = SqlUpdate
   , sqlUpdateWhere  :: [SqlCondition]
   , sqlUpdateSet    :: [(SQL,SQL)]
   , sqlUpdateResult :: [SQL]
-  , sqlUpdateWith   :: [(SQL, SQL)]
+  , sqlUpdateWith   :: [(SQL, SQL, Materialized)]
   }
 
 data SqlInsert = SqlInsert
   { sqlInsertWhat       :: SQL
   , sqlInsertOnConflict :: Maybe (SQL, Maybe SQL)
   , sqlInsertSet        :: [(SQL, Multiplicity SQL)]
   , sqlInsertResult     :: [SQL]
-  , sqlInsertWith       :: [(SQL, SQL)]
+  , sqlInsertWith       :: [(SQL, SQL, Materialized)]
```

Comment:
This looks like an ugly API break. Maybe it would be better to add `sqlInsertWithMaterialized` so that the existing code doesn't have to be changed with the new constructor?

Reply:
You mean new field? Not that would mean more handling code of fields that are mutually exclusive - not sure we want this. @arybczak any thoughts?

Reply:
It's fine to break API, we're the only ones using the library anyway.

Reply:
btw - should we really expose those constructors anyway (given there are helper functions?)

Reply:
In principle we can hide them, but let's not do that in this PR.